packages/chain-adapters/src/utxo/utxoSelect/index.ts

```diff
@@ -4,20 +4,31 @@ import split from 'coinselect/split'
 
 export type UTXOSelectInput = {
   utxos: unchained.bitcoin.Utxo[]
+  from?: string
   to: string
   value: string
   satoshiPerByte: string
   opReturnData?: string
   sendMax: boolean
 }
 
+type SanitizedUTXO = Omit<unchained.bitcoin.Utxo, 'value'> & { value: number }
+
 /**
  * Returns necessary utxo inputs & outputs for a desired tx at a given fee with OP_RETURN data considered if provided
  *
  * _opReturnData is filtered out of the return payload as it is added during transaction signing_
  */
 export const utxoSelect = (input: UTXOSelectInput) => {
-  const utxos = input.utxos.map((x) => ({ ...x, value: Number(x.value) }))
+  const utxos = input.utxos.reduce<SanitizedUTXO[]>((acc, utxo) => {
+    const sanitizedUtxo = { ...utxo, value: Number(utxo.value) }
+
+    // If input contains a `from` param, the intent is to only keep the UTXOs from that address
+    // so we can ensure the send address is the one we want
+    // This doesn't do any further checks, so error-handling should be done by the caller e.g `buildSendTransaction` callsites
+    return [...acc, ...(input.from && utxo.address !== input.from ? [] : [sanitizedUtxo])]
```

Comment:
this can actually be a decent amount of data and is on the hot path, may want to `acc.push(...` here